tests/suites/test_suite_aes.xex.data

```diff
@@ -0,0 +1,3599 @@
+
+AES-128-XEX Encrypt NIST XTSTestVectors #231
+aes_encrypt_xex:"9b7f94044bd13ebad385bf4fd9c0bc18280bee7f1c43c2ed29352f4a0f4eea18":"e2508de947e3c32f66d511f5e4f596ff":"7325bbc4f480001a0406da6af0bd34909f59c57ab99d718b91e6686848cb9deb":"6ba938d27feca55d18ba2a1af546e6234bd89ab54596a8bc3f6fdb3e719d2887"
+
+AES-128-XEX Encrypt NIST XTSTestVectors #232
+aes_encrypt_xex:"a70f0fc7f92db7a64cb6d029b2e8d127021416415e16c655f253654daae9e1e5":"dd4001dacfad534677066c4f59ecf734":"318506eb10a753b1cc920df731f7c51d4ef50bd1d6cde1222b414cce212e727d":"a7092e32f1a1e7a6cd89c79e5dd99377a52d2f421591615fb4d928d8c019c7c1"
+
+AES-128-XEX Encrypt NIST XTSTestVectors #233
+aes_encrypt_xex:"db3f026308b712911b8d5418b71eeff753a8614eeb6495fd6d06da1d3a939169":"db8a6c216a01f4a93ee7dd4d7818451e":"7d87ee2b6f8e0dcb55f67eff96e9182688cc0261d7ce3409dbb4a38e2002309c":"4fa94efd047d6254406863d92fd7c05245e9c3335360926453c9f935e8acb0fa"
+
+AES-128-XEX Encrypt NIST XTSTestVectors #234
+aes_encrypt_xex:"b8d820058c20a34b55aec16bd9ff10a9f9de3c071caf2217a4422c0bd39a2fc9":"b97d9c8bd7f5a503233c11164c252e5b":"d085378688c0836a76e39285f5e3a786748d9f97074caf3c65c25696ecf1f23e":"a9697d5047abf5e31517614cf0e41d27ede31aeedb73a32a22133329990eb6ac"
+
+AES-128-XEX Encrypt NIST XTSTestVectors #235
+aes_encrypt_xex:"c1bdf06902747962e7453e201e4011fd03974eaff8e2758ad47e4f60ece2aec9":"27299656b85340e7397aaf20e795c28e":"9c7c618d681c032ef10f2b554d3a5067084d68570377bf496b953122dcb0b333":"fa2b8d449536fc11c117702de757a0b4790745cb04ef3e0c6cf4e166b177c08c"
+
+AES-128-XEX Encrypt NIST XTSTestVectors #236
+aes_encrypt_xex:"c5daf1a726f3419eaa32936e7e12e1a4ac6685487986c9a38470ce23e64ab07a":"4ed794b26083938f7c4b51f6d5fdc2f1":"e25141cdd9ae22c378b864515fb8d7f2320aa24c3e747d1b566675362e2c8ead":"1357df892b91f7c2bf7f7ebf2b6fa3f6783effe36aa66d53ed21a50156b8c739"
+
+AES-128-XEX Encrypt NIST XTSTestVectors #237
+aes_encrypt_xex:"0560ec1e3469e4eea9c04a6a5baa39c1bceb64ce9a83fbce8f240ad41ce63013":"3adcadd94ac5d385bbeadb7e12b56d8b":"88ac15329974a9cbd8a5d6f370bbe7fe31705e82f24663917c27be319bccc756":"9c2bbec2626638dba8e7d4a5d47424154912fd9c18b72adeb5ee85815cc19b6a"
+
+AES-128-XEX Encrypt NIST XTSTestVectors #238
+aes_encrypt_xex:"725c6cb3a8f4ab8c524f1c8a5626029e85f11a04d8593e056387ef493ac6ba25":"6235da2b9673518df8a356569ec9bb63":"6c68d2f3ef6241ecd0610654a58fc68c2b5b60da73d7380ac33638b42302063d":"86868e16193f58c9cb9ed42362e15efad1baa79a8faab6947fc0adcb48bdea5d"
+
+AES-128-XEX Encrypt NIST XTSTestVectors #239
+aes_encrypt_xex:"86bd097c948e671deba9cdab9cf7611235eefe5da54ce47eb5c7cbea6c583af7":"0753ba896e5733ed3e1aaa93afeed7af":"779614513f7414929df2bec1d0989dad294c5dc80ec630f56e76b5490ebc43e3":"88fec29008dac2f894bf53364701eb472ed57c542df2ec02942c8b87c201c898"
+
+AES-128-XEX Encrypt NIST XTSTestVectors #240
+aes_encrypt_xex:"b3ede67af12bc4bb90e16
```

Comment:
300 test cases is just a little bit over the top. One test case per length among {16,17,24,31,32,48,512} and per key size would be enough. (I added 24 and 31 because I think half-a-block and block-minus-one are more likely to reveal an off-by-one error somewhere.) This also goes for XTS.

Reply:
I deleted all the XEX tests.

I would like to hear also from @yanesca as to which subset of the test vectors we should run with before I remove any.

What would you both think we if removed all the NIST test vectors and use only the IEEE test vectors?